Formula/opendht.rb

```diff
@@ -0,0 +1,39 @@
+class Opendht < Formula
+  desc "C++14 Distributed Hash Table implementation"
+  homepage "https://github.com/savoirfairelinux/opendht"
+  url "https://github.com/savoirfairelinux/opendht/archive/refs/tags/2.3.4.tar.gz"
+  sha256 "3c902f30bae21c6ac1ff08417f01c9a6ad252db32afac72508ed321d8478c5ab"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "argon2"
+  depends_on "asio"
+  depends_on "gnutls"
+  depends_on "msgpack-cxx"
```

Comment:
From linkage test, `asio` and `msgpack-cxx` aren't linked, so I want to check if these are build-time/header-only
```
Dependencies with no linkage:
  asio
  msgpack-cxx
```

If they aren't used at runtime, it would be better to mark them `=> :build`

Reply:
`msgpack-cxx` is headed-only.

Reply:
Actually, taking a look, they may be necessary to use `opendht` headers.

Though, ArchLinux only lists them as needed during make https://archlinux.org/packages/community/x86_64/opendht/